src/sentry/seer/anomaly_detection/store_data.py

```diff
@@ -57,16 +58,28 @@ def _get_start_and_end_indices(data: list[TimeSeriesPoint]) -> tuple[int, int]:
     return start, end
 
 
-def send_historical_data_to_seer(alert_rule: AlertRule, project: Project) -> AlertRuleStatus:
+def send_historical_data_to_seer(
+    alert_rule: AlertRule,
+    project: Project,
+    snuba_query=None,
+    event_types: list[SnubaQueryEventType.EventType] | None = None,
+) -> AlertRuleStatus:
     """
     Get 28 days of historical data and pass it to Seer to be used for prediction anomalies on the alert.
     """
-    snuba_query = SnubaQuery.objects.get(id=alert_rule.snuba_query_id)
+    if not snuba_query:
+        snuba_query = SnubaQuery.objects.get(id=alert_rule.snuba_query_id)
     window_min = int(snuba_query.time_window / 60)
-    dataset = get_dataset(snuba_query.dataset)
+    dataset = get_dataset_from_label(snuba_query.dataset)
     query_columns = get_query_columns([snuba_query.aggregate], snuba_query.time_window)
+    if not event_types:
+        event_types = snuba_query.event_types
```

Comment:
is `snuba_query.event_types` always set? (we may want to provide a default if this is required or throw an exception if it's a required field)

Reply:
Yes a snuba query will always have event types on it, but I changed this function to accept both `snuba_query` and `event_types` because this is happening during an alert rule _update_ - to avoid a bad state, we don't want to save the changes until we've successfully sent the data to Seer, so I can't pull the snuba query off of the alert rule (it may have changed) and the event types on the snuba query get updated separately